src/category_theory/limits/shapes/biproducts.lean

```diff
@@ -1299,13 +1309,49 @@ end
   biprod.lift f g ≫ biprod.desc h i = f ≫ h + g ≫ i :=
 by simp [biprod.lift_eq, biprod.desc_eq]
 
-
 lemma biprod.map_eq [has_binary_biproducts C] {W X Y Z : C} {f : W ⟶ Y} {g : X ⟶ Z} :
   biprod.map f g = biprod.fst ≫ f ≫ biprod.inl + biprod.snd ≫ g ≫ biprod.inr :=
 by apply biprod.hom_ext; apply biprod.hom_ext'; simp
 
 end
 
+section
+variables {X Y : C} (f g : X ⟶ Y)
+
+/-- The existence of binary biproducts implies that there is at most one preadditive structure. -/
+lemma biprod.add_eq_lift_id_desc [has_binary_biproduct X X] :
+  f + g = biprod.lift (𝟙 X) (𝟙 X) ≫ biprod.desc f g :=
+by simp
+
+/-- The existence of binary biproducts implies that there is at most one preadditive structure. -/
+lemma biprod.add_eq_lift_desc_id [has_binary_biproduct Y Y] :
+  f + g = biprod.lift f g ≫ biprod.desc (𝟙 Y) (𝟙 Y) :=
+by simp
+
+end
+
+
 end preadditive
 
-end category_theory.limits
+end limits
+
+open category_theory.limits
+
+local attribute [ext] preadditive
+
+/-- The existence of binary biproducts implies that there is at most one preadditive structure. -/
+instance subsingleton_preadditive_of_has_binary_biproducts {C : Type u} [category.{v} C]
+  [has_zero_morphisms C] [has_binary_biproducts C] : subsingleton (preadditive C) :=
+subsingleton.intro $ λ a b,
+begin
+  ext X Y f g,
+  have h₁ := @biprod.add_eq_lift_id_desc _ _ a _ _ f g
+    (by convert (infer_instance : has_binary_biproduct X X)),
+  have h₂ := @biprod.add_eq_lift_id_desc _ _ b _ _ f g
+    (by convert (infer_instance : has_binary_biproduct X X)),
+  refine h₁.trans (eq.trans _ h₂.symm),
```

Comment:
Can you just inline the lemmas from the section above?

Reply:
I think it would end up quite long and painful because I would have to make `a` and `b` known to the typeclass system one after the other.